src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1247,20 +1249,25 @@ BlankNode Visitor::visitTypesafe(Parser::BlankNodeContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-template <typename Out, typename FirstContext, typename... Context>
-Out Visitor::visitAlternative(FirstContext ctx, Context... ctxs) {
+template <typename Out, bool isRecursive, typename FirstContext,
+          typename... Context>
+Out Visitor::visitAlternative(FirstContext* ctx, Context*... ctxs) {
+  if constexpr (!isRecursive) {
+    int validContexts =
```

Comment:
maybe better `numValidContexts` then you don't need to comment the whole `...` business.